### PR TITLE
A better way to handle remote content-type, fix etag

### PR DESCRIPTION
**Pull request by n0vad3v**
* This PR will have a better way to determine `content-type` when using remote mode
  * Instead of relying on `content-type` header returned from remote, it will download the remote content and check for MIME type, this will solve problem in issue https://github.com/webp-sh/webp_server_go/issues/215 and https://github.com/webp-sh/webp_server_go/issues/213, however, risks at when requesting something big, it will waste some traffic to fetch it. (Consider remote has `/path/to/image.jpg` and `/path/to/10GB.bin` and requests are made to  the latter one).
  * Maybe we can add more checks here to mitigate some risks.
* Added back etag value used for caching remote original image, which is being removed in https://github.com/webp-sh/webp_server_go/pull/194
  * Now when fetching remote image, contents in `remote-raw` will be `Sha1Path(reqURIwithQuery) + "-etag-" + Sha1Path(etagValue)`, example: `/home/webp_server/remote-raw/378e740ca56144b7587f3af9debeee544842879a-etag-123e740ca56333b7587f3af9debeee5448428123`.
  *  As on Linux the maximum length for a file name is 255 bytes. The maximum combined length of both the file name and path name is 4096 bytes. This shouldn't give us any problems.

### CODE REVIEW

**Comment by bugfest** [comment]
HI @n0vad3v, I've being thinking about how to improve the cache; here you're using `go-cache` for the in-memory locking but I was wondering if we could take the opportunity to use it thru the `gocache` wrapper instead: https://github.com/eko/gocache#memory-using-go-cache

That way we could extend the server to use other caching services in the future more easily.

File: helper.go [hunk 1]
```diff
@@ -65,6 +65,19 @@ func imageExists(filename string) bool {
 		// means something wrong in exhaust file system
 		return false
 	}
+
+	// Check if there is lock in cache, retry after 1 second
+	if _, found := WriteLock.Get(filename); found {
```
Comment (BennyThink) [inline helper.go:70]: for(var i;i<=1;i++){
		// try to get lock...
		// break and return if necessary
	}

**Review by BennyThink — CHANGES_REQUESTED** [review]

**Comment by n0vad3v** [comment]
@bugfest 
> I was wondering if we could take the opportunity to use it thru the gocache wrapper instead 

Agree with you, however I'm keep using this in this PR right now as we only have this one item to cache and trying to fix the problem in issue: https://github.com/webp-sh/webp_server_go/issues/213

I've also edited https://github.com/webp-sh/webp_server_go/issues/217 to add this potential feature.

**Review by bugfest — COMMENTED** [review]
best commit message of all times

**Review by BennyThink — APPROVED** [review]

**Pull request merged**